Modules/Shop/Http/Services/ShopService.php

```diff
@@ -417,7 +414,7 @@ public function getProduct(int $productId): array
 
         if ($productDetails) {
             if ($user) {
-                $wishList = Wishlist::where('product_id', $productDetails->id)->where('user_id', $user->id)->where('cart_id', null)->first();
+                $wishList = Product::find($productDetails->id)->productWishlist()->where('user_id', $user->id)->where('cart_id', null)->first();
```

Comment:
Get the relation productWishlist when you get the $products and instead of this query use
$wishList = $productDetails->productWishlist()->where('user_id', $user->id)->where('cart_id', null)->first();